### PR TITLE
Use PocketIc API V2

### PR DESCRIPTION
**Pull request by michael-weigelt**

### CODE REVIEW

File: pocket_ic/pocket_ic_server.py [hunk 1]
```diff
@@ -29,7 +30,7 @@ class PocketICServer:
     def __init__(self) -> None:
         # Attempt to start the PocketIC server if it's not already running.
         pid = os.getpid()
-        os.system(f"{BINARY_NAME} --pid {pid} &")
+        os.system(f"{BINARY_NAME} --pid {pid} --log-level WARN &")
```
[the remark this answers is not in the document]
Comment (fxgst) [reply pocket_ic/pocket_ic_server.py]: But there's an MR coming that will change this behavior anyway, and maybe we then do not need to pass that flag at all

File: pocket_ic/pocket_ic.py [hunk 2]
```diff
@@ -311,7 +292,12 @@ def create_and_install_canister_with_candid(
     def _get_ok_reply(self, request_result):
         if "Ok" in request_result:
             if "Reply" in request_result["Ok"]:
-                return request_result["Ok"]["Reply"]
+                result = request_result["Ok"]["Reply"]
+                maybe_candid = base64.b64decode(result)
+                # if we have a non-candid byte array, return that without decoding
+                if b'DIDL' in maybe_candid:
```
[the remark this answers is not in the document]
Comment (fxgst) [reply pocket_ic/pocket_ic.py:298]: Or
```suggestion
                if maybe_candid.startswith(b'DIDL'):
```

File: pocket_ic/pocket_ic.py [hunk 3]
```diff
@@ -23,16 +23,14 @@ def __init__(self) -> None:
...
-            body (Any): the payload for the IC request
+    def instance_get(self, endpoint):
+        """HTTP get requests for instance endpoints"""
+        return self.server.instance_get(endpoint, self.instance_id)
 
-        Returns:
-            Any: a JSON encoded result, if the request contains a response
-        """
-        return self.server.send_request_to_instance(self.instance_id, body)
+    def instance_post(self, endpoint, body):
```
Comment (fxgst) [inline pocket_ic/pocket_ic.py:31]: this as well?

File: tests/pocket_ic_test.py [hunk 4]
```diff
@@ -58,7 +59,7 @@ def test_canister_exists_negative(self):
     def test_cycles_balance(self):
         canister_id = self.pic.create_canister()
         self.pic.add_cycles(canister_id, 6_666)
-        self.assertEqual(self.pic.get_cycles_balance(canister_id), 6_666)
+        self.assertEqual(self.pic.get_cycles_balance(canister_id)["cycles"], 6_666)
```
Comment (fxgst) [inline tests/pocket_ic_test.py:62]: Either we have to remove the type hint of the return type of `get_cycles_balance` (it's claiming it's returning an `int`), or we do the `["cycles"]` in there, which I think is preferable.

File: pocket_ic/pocket_ic_server.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fxgst) [reply pocket_ic/pocket_ic_server.py]: I see you just removed that in a new commit 👍🏻

File: pocket_ic/pocket_ic.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (michael-weigelt) [reply pocket_ic/pocket_ic.py:298]: Agreed

**Review by fxgst — APPROVED** [review]

**Pull request merged**